### PR TITLE
Added a recipe type for reactor fuel & use tags for redstone catalyst

### LINKED ISSUE #108: [1.19.2 Feature Request] Tag (or config) to add more accepted items to Reactors
#### Feature description:
I'd personally prefer a tag, but it'd be nice if both slots in a Reactor could accept different items based on what was tagged/configured.

Having it apply to at least Carbon would be good imo.

Ideas for the tag's name:
Carbon Slot: carbon
Redstone Slot: stabilizer (also changes the hover label for redstone)
Uraninite Slot: fuel

Adding a section to the config to add more solid coolants would be great too. There's one for Thermo Heat Sources, so it's a bit odd there isn't one for Coolants too.

#### Why you think this feature should be added:
It'd make Reactors a bit better and allow more flexibility in custom modpacks.

(also, the CurseForge page doesn't have a quick link for this issue tracker. Would be nice if it was added).

### PR DESCRIPTION
**Pull request by shartte**
Fixes #108

### CODE REVIEW

File: src/main/java/owmii/powah/block/reactor/ReactorTile.java [hunk 1]
```diff
@@ -324,10 +334,6 @@ public void onPlaced(Level world, BlockState state, @Nullable LivingEntity place
 
     public void demolish(Level world) {
         this.builder.demolish(world);
-        while (this.fuel.getTicks() >= 100) {
```
Comment (Technici4n) [inline src/main/java/owmii/powah/block/reactor/ReactorTile.java:327]: What about this? Is the fuel lost?

File: src/generated/resources/data/powah/recipes/reactor_fuels/uraninite.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "powah:reactor_fuel",
```
Comment (Technici4n) [inline src/generated/resources/data/powah/recipes/reactor_fuels/uraninite.json]: This is the perfect use case for a DataMap

File: src/main/java/owmii/powah/block/reactor/ReactorTile.java (hunk 1 above)
Comment (shartte) [reply src/main/java/owmii/powah/block/reactor/ReactorTile.java:327]: Yes

File: src/generated/resources/data/powah/recipes/reactor_fuels/uraninite.json (hunk 2 above)
Comment (shartte) [reply src/generated/resources/data/powah/recipes/reactor_fuels/uraninite.json]: A what map? :-D

**Comment by shartte** [comment]
@Technici4n is this better? 😓 
https://github.com/Technici4n/Powah/pull/136/files#diff-71617cb2ce82fc9a2e1d468bffac9eea927d7e96939c5f8ea6b6225b2964c4dfR17

**Pull request merged**